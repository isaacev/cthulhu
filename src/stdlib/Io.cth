intrinsic {
  fn php_print(Str) -> ();
  fn read_argv() -> [Str];
}

#[inline]
pub fn print(str: Str) -> () {
  php_print(str)
}

#[inline]
pub fn println(str: Str) -> () {
  php_print(str ++ "\n")
}

pub fn argv() -> [Str] {
  read_argv()
}
