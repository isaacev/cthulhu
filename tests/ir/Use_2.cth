use ::Io;

mod Foo {
  pub mod Bar {
    use ::Io;

    fn hello() -> () {
      Io::println("hello");
    }
  }
}

#[entry]
fn main() -> () {
  Foo::Bar::hello();
}
