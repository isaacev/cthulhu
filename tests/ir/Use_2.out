
  UNKNOWN FIELD  -------------------------------------------------------------

  Reference to an unknown field in a namespace.

     : Use_2
  15 |   Foo::Bar::hello();
     |             ^^^^^

